### PR TITLE
Fix handling of non-ASCII characters in new `*_url` filters

### PR DESCRIPTION
**Pull request by pathawks**
How/where should normalization be handled? I would love if the filters could also be responsible for normalization.

### CODE REVIEW

**Comment by parkr** [comment]
Punycode, you are the worst. Encoding URL's is notorious. I think this could be a fix for after 3.3 just because it's so gnarly.

Does Ruby stdlib have anything for this?

File: jekyll.gemspec [hunk 1]
```diff
@@ -38,4 +38,5 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency('jekyll-sass-converter', '~> 1.0')
   s.add_runtime_dependency('jekyll-watch', '~> 1.1')
   s.add_runtime_dependency("pathutil", "~> 0.9")
+  s.add_runtime_dependency('addressable', '~> 2.4')
```
Comment (pathawks) [inline jekyll.gemspec]: **Addressable** was already being installed implicitly, so this is not _actually_ adding a new dependency.

**Comment by pathawks** [comment]
> I think this could be a fix for after 3.3 just because it's so gnarly.

Oh please no. I feel like plugins won't be able to rely on this filter until this is resolved.

> Does Ruby stdlib have anything for this?

I don't think so, and that's why [Addressable](https://github.com/sporkmonger/addressable) exists. It was already an implicit dependency, so I don't see a problem using it here.

File: jekyll.gemspec (hunk 1 above)
Comment (parkr) [inline jekyll.gemspec]: Doesn't look like it's a dependency of jekyll, but it a dependency of `jekyll-sitemap` per a Gemfile.lock of mine using both. So it is technically not used yet.

Comment (pathawks) [inline jekyll.gemspec]: Ok, that makes sense.

**Comment by pathawks** [comment]
Travis likes it! ✅

**Comment by parkr** [comment]
I'm OK with this. @jekyll/stability, any thoughts?

**Comment by parkr** [comment]
@jekyllbot: merge +bug

**Pull request merged**